From ad3a0be61c020103bb95149e361b3cda5a3facbc Mon Sep 17 00:00:00 2001
From: darkyzhou <me@zqy.io>
Date: Wed, 10 Dec 2025 00:31:01 +0800
Subject: [PATCH 08/21] AOSCOS: build: link libm

the math library (libm) needs to be explicitly linked to resolve
symbols like nan() and log2(). This patch adds "m" to the default_libs
configuration for Linux builds.

This fixes linker errors such as:
  ld.lld: error: undefined symbol: nan
  ld.lld: error: undefined symbol: log2

Signed-off-by: darkyzhou <me@zqy.io>
---
 build/config/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/config/BUILD.gn b/build/config/BUILD.gn
index 96085b4..9a2793d 100644
--- a/build/config/BUILD.gn
+++ b/build/config/BUILD.gn
@@ -214,6 +214,7 @@ config("default_libs") {
       "dl",
       "pthread",
       "rt",
+      "m",
     ]
   }
 }
-- 
2.54.0

